src/party.cpp

```diff
@@ -375,11 +375,7 @@ void Party::updateVocationsList()
 	}
 
 	size_t size = vocationIds.size();
-	if (size > 1) {
-		extraExpRate = static_cast<float>(size * (10 + (size - 1) * 5)) / 100.f;
-	} else {
-		extraExpRate = 0.20f;
-	}
+	sharedExpBonus = size > 1 ? static_cast<float>(size * (10 + (size - 1) * 5)) / 100.f : 0.20f;
```

Comment:
Replace this ternary with `std::min(0.2f, <big function here>)` .

I think you don't even need the float cast since division by float is always float, too. And I wouldn't even use `float` since it is casted to `double` everywhere.

Reply:
Should I change it's data type to ``double`` then?

I think the ``#include "monsters.h"`` is because someone was probably going to add a missing check regarding the shared experience mechanism, in which it should not apply the bonus to the ``shareExperience`` variable in ``Party::shareExperience`` in case the target creature (a monster) does not yield atleast ``20`` experience points.

I'll remove it nonetheless, but I may research this with more caution in vanilla, and I'll make a PR for it. But I think it is something like this:
```C++
void Party::shareExperience(uint64_t experience, Creature* source/* = nullptr*/)
{
	uint64_t shareExperience;
	if (source) {
		MonsterType* monsterType = g_monsters.getMonsterType(source:getName());
		if (monsterType && monsterType->getExperience() < 20) {
			shareExperience = experience
		}
	} else {
		shareExperience = static_cast<uint64_t>(std::ceil((static_cast<double>(experience) * (sharedExpBonus + 1)) / (memberList.size() + 1)))
	}

	for (Player* member : memberList) {
		member->onGainSharedExperience(shareExperience, source);
	}
	leader->onGainSharedExperience(shareExperience, source);
}
```

Reply:
Yes, sorry, I mistyped `min` for `max`, it should be the latter.

You can safely change to `double`, as far as I am concerned there are no drawbacks.